server/services/store/sqlstore/initialize.go

```diff
@@ -28,60 +49,88 @@ func (s *SQLStore) InitializeTemplates() error {
 // changes
-func (s *SQLStore) importInitialTemplates() error {
-	/*
-		s.logger.Debug("importInitialTemplates")
-		blocksJSON := initializations.MustAsset("templates.json")
-
-		var archive model.Archive
-		err := json.Unmarshal(blocksJSON, &archive)
-		if err != nil {
-			return err
-		}
+func (s *SQLStore) importInitialTemplates(db sq.BaseRunner, teamIDs []string) error {
+	s.logger.Debug("importInitialTemplates")
+	blocksJSON := initializations.MustAsset("templates.json")
 
-		// ToDo: loop over teams
+	var archive model.Archive
+	err := json.Unmarshal(blocksJSON, &archive)
+	if err != nil {
+		return err
+	}
 
-		globalContainer := store.Container{
-			TeamID: "0",
-		}
+	for _, teamID := range teamIDs {
+		for _, board := range archive.Boards {
+			boardBlocks := []model.Block{}
+			newBoardID := utils.NewID(utils.IDTypeBoard)
+			for _, block := range archive.Blocks {
+				if block.ParentID == board.ID || block.RootID == board.ID {
+					boardBlocks = append(boardBlocks, block)
+				}
+			}
 
-		s.logger.Debug("Inserting blocks", mlog.Int("block_count", len(archive.Blocks)))
-		for i := range archive.Blocks {
-			s.logger.Trace("insert block",
-				mlog.String("blockID", archive.Blocks[i].ID),
-				mlog.String("block_type", archive.Blocks[i].Type.String()),
-				mlog.String("block_title", archive.Blocks[i].Title),
-			)
-			err := s.InsertBlock(globalContainer, &archive.Blocks[i], "system")
+			boardBlocks = model.GenerateBlockIDs(boardBlocks)
+			updatedBoardBlocks := []model.Block{}
+
+			for _, block := range boardBlocks {
+				updatedBlock := block
+				if updatedBlock.ParentID == updatedBlock.ID {
+					updatedBlock.ParentID = newBoardID
+				}
+				if updatedBlock.RootID == updatedBlock.ID {
+					updatedBlock.RootID = newBoardID
+				}
+				updatedBlock.BoardID = newBoardID
+				updatedBoardBlocks = append(updatedBoardBlocks, updatedBlock)
+			}
+			board.ID = newBoardID
+			board.Type = model.BoardTypeOpen
+			board.TeamID = teamID
+			board.IsTemplate = true
+
+			for _, block := range updatedBoardBlocks {
+				err := s.insertBlock(db, &block, "system")
+				if err != nil {
+					return err
+				}
+			}
+			_, err := s.insertBoard(db, &board, "system")
 			if err != nil {
 				return err
 			}
 		}
-	*/
+	}
 
 	return nil
 }
 
-// ToDo: as part of the change in the above method,
-// isInitializationNeeded should be something like:
-// getTeamsThatNeedInitialization() ([]string, error)
-// and return the team IDs that have no boards
-
-// isInitializationNeeded returns true if the blocks table is empty.
-func (s *SQLStore) isInitializationNeeded() (bool, error) {
+// getTeamsThatNeedInitialization returns a list of teams without templates
+func (s *SQLStore) getTeamsThatNeedInitialization(db sq.BaseRunner) ([]string, error) {
 	query := s.getQueryBuilder(s.db).
-		Select("count(*)").
-		From(s.tablePrefix + "blocks").
-		Where(sq.Eq{"COALESCE(channel_id, '0')": "0"})
-
-	row := query.QueryRow()
+		Select("T.Id").
```

Comment:
This won't allow us to update existing templates as we need to do for growth spike. I'm not sure if should handle it here or cherrypick it later on from `main` after the growth spike.